pkg/operator/sync.go

```diff
@@ -528,6 +538,55 @@ func (optr *Operator) applyManifests(config *renderConfig, paths manifestPaths)
 	return nil
 }
 
+// safetySyncControllerConfig is a special case render of the controllerconfig that we run when
+// we need to keep syncing controllerconfig, but something (like daemon sync failing) prevents
+// us from getting to the actual controller sync
+func (optr *Operator) safetySyncControllerConfig(config *renderConfig) error {
+	glog.Infof("Performing safety controllerconfig sync")
+
+	// If we have an existing controllerconfig, we might be able to keep rendering
+	existingCc, err := optr.ccLister.Get("machine-config-controller")
+	if err != nil {
+		return err
+	}
+
+	// If there is an existing config, but we didn't render it (e.g. we are in the middle of an upgrade)
+	// we can't render a new one here, it won't succeed because the existing controller won't touch it
+	// and we'll time out waiting.
+	if existingCc.Annotations[daemonconsts.GeneratedByVersionAnnotationKey] != version.Raw {
+		return fmt.Errorf("Our version (%s) differs from controllerconfig (%s), can't do 'safety' controllerconfig sync until controller is updated", version.Raw, existingCc.Annotations[daemonconsts.GeneratedByVersionAnnotationKey])
+	}
+
+	// If we made it here, we should be able to sync controllerconfig, and the existing controller should handle it
+	return optr.syncControllerConfig(config)
+}
+
+// syncControllerConfig is NOT currently intended to be called as a separate sync, it is called as part of syncMachineConfigController.
```

Comment:
Sorry, just to make sure I understand this comment, you are saying that ControllerConfig generally speaking should not be synced independently of the syncMachineConfigController loop, but this is ok here since you have an addition check for `GeneratedByVersionAnnotationKey` above?

Also this is a subset of the syncMachineConfigController func right? Out of curiosity why not split that function into two and call syncControllerConfig in syncMachineConfigController? I don't mind either way but it might make it easier to maintain

Reply:
Correct, I rephrased the comment to hopefully make it more clear. (If I failed I'll try again :smile: ) 

> why not split that function into two and call syncControllerConfig in syncMachineConfigController
Hah, originally I did call that from syncMachineConfigController (you can even see I referenced it in the comment) but I accidentally scraped it off when I stripped off my logging/debug commit. Fixed.